ethers-solc/src/artifacts.rs

```diff
@@ -708,13 +727,22 @@ impl<'a> fmt::Display for OutputDiagnostics<'a> {
         } else {
             f.write_str("Compiler run successful")?;
         }
-        for err in self.errors {
-            // Do not log any ignored error codes
-            if let Some(error_code) = err.error_code {
-                if !self.ignored_error_codes.contains(&error_code) {
-                    writeln!(f, "\n{}", err)?;
+        for err in &self.compiler_output.errors {
+            let is_ignored = err.error_code.as_ref().map_or(false, |code| {
+                if let Some(source_location) = &err.source_location {
+                    // we ignore spdx and contract size warnings in test
+                    // files. if we are looking at one of these warnings
+                    // from a test file we skip
+                    if self.is_test(&source_location.file)
+                        && (*code == 1878 || *code == 5574) {
```

Comment:
Not super keen on this, but I am not entirely sure what direction to go in with fully configurable error codes based on types either in terms of DX or configuration. Is this fine for now or does anyone have any pointers on how to make this more elegant?